web-access-app/src/main/java/edu/unc/lib/boxc/web/access/controllers/FullRecordController.java

```diff
@@ -238,12 +238,14 @@ public String getFullRecord(String pidString, Model model, HttpServletRequest re
             boolean pdfViewerNeeded = false;
 
             if (!imageViewerNeeded) {
+                String pdfMimeTypeRegx = "application/(x-)?pdf";
                 // Check if primary object is a pdf
-                pdfViewerNeeded = accessCopiesService.hasViewablePdf(briefObject, principals);
+                pdfViewerNeeded = accessCopiesService.hasDatastreamContent(briefObject, principals, pdfMimeTypeRegx);
```

Comment:
I'm finding the various slightly different helper methods kind of hard to follow and i'm not sure why audio behaves differently than pdfs? 

How about using `getDatastreamPid` for both pdfs and audio. `getDatastreamPid` will return the pid of the primary object if it has one, and if not then it will do the looking up children stuff it already does. Then for model attributes you could have "pdfViewerNeeded", "audioPlayerNeeded", and "viewerPid", where the latter is the pid of the file to display regardless of viewer. Or you could combine all the flag attributes into one, and have "viewerType" with values "uv", "pdf", "audio", and "viewerPid". We only show one viewer at a time so we don't really need to allow for multiple viewer attributes to be set.

So
```
String viewerType = null;
String viewerPid = null;
boolean imageViewerNeeded = accessCopiesService.hasViewableFiles(briefObject, principals);
if (imageViewerNeeded) {
    viewerType = "uv";
} else {
    // Check for PDF to display
    viewerPid = accessCopiesService.getDatastreamPid(briefObject, principals, pdfMimeTypeRegx);
    if (viewerPid != null) {
        viewerType = "pdf";
    } else {
        // Check for viewable audio file
        viewerPid = accessCopiesService.getDatastreamPid(briefObject, principals, audioMimeTypeRegx);
        if (viewerPid != null) {
            viewerType = "audio";
        }
    }
}
model.addAttribute("viewerType", viewerType);
model.addAttribute("viewerPid", viewerPid);
```